module/os/linux/zfs/zpl_xattr.c

```diff
@@ -1148,7 +1148,7 @@ zpl_init_acl(struct inode *ip, struct inode *dir)
 		if (IS_ERR(acl))
 			return (PTR_ERR(acl));
 		if (!acl) {
-			ip->i_mode &= ~current_umask();
+			ITOZ(ip)->z_mode = ip->i_mode &= ~current_umask();
```

Comment:
```suggestion
			ITOZ(ip)->z_mode = (ip->i_mode &= ~current_umask());
```

I'd suggest adding parenthesis here to make it absolutely clear the `&=` is evaluated first.  Or just move the `ITOZ(ip)->z_mode` to it's own line.